warehouse/integrations/deltalake-native/deltalake_native.go

```diff
@@ -0,0 +1,1367 @@
+func (d *Deltalake) partitionQuery(tableName string) (string, error) {
+	if !d.EnablePartitionPruning {
+		return "", nil
+	}
+
+	var (
+		query            string
+		partitionColumns []string
+		rows             *sql.Rows
+		err              error
+	)
+
+	query = fmt.Sprintf(`SHOW PARTITIONS %s.%s;`, d.Namespace, tableName)
+
+	if rows, err = d.DB.Query(query); err != nil {
+		if strings.Contains(err.Error(), partitionNotFound) {
+			return "", nil
+		}
+		return "", fmt.Errorf("executing fetching partitions: %w", err)
+	}
+	defer func() { _ = rows.Close() }()
+
+	if partitionColumns, err = rows.Columns(); err != nil {
+		return "", fmt.Errorf("scanning partition columns: %w", err)
+	}
+
+	if !isPartitionedByEventDate(partitionColumns) {
+		return "", nil
+	}
+
+	firstEvent, lastEvent := d.Uploader.GetFirstLastEvent()
+	dateRange := warehouseutils.GetDateRangeList(firstEvent, lastEvent, "2006-01-02")
+	if len(dateRange) == 0 {
+		return "", nil
+	}
+
+	dateRangeString := warehouseutils.JoinWithFormatting(dateRange, func(idx int, str string) string {
+		return fmt.Sprintf(`'%s'`, str)
+	}, ",")
+	partitionQuery := fmt.Sprintf(`CAST ( MAIN.event_date AS string) IN (%s) AND`, dateRangeString)
+
+	return partitionQuery, nil
+}
+
+// schemaExists checks it schema exists or not.
+func (d *Deltalake) schemaExists(schemaName string) (bool, error) {
+	var (
+		query  string
+		err    error
+		schema string
+	)
+
+	query = fmt.Sprintf(`SHOW SCHEMAS LIKE '%s';`, schemaName)
+
+	err = d.DB.QueryRow(query).Scan(&schema)
+	if err == sql.ErrNoRows {
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("schema exists: %w", err)
+	}
+
+	exists := schema == schemaName
+	return exists, nil
+}
+
+func (d *Deltalake) createSchema() error {
+	var (
+		query string
+		err   error
+	)
+
+	query = fmt.Sprintf(`CREATE SCHEMA IF NOT EXISTS %s;`, d.Namespace)
+
+	if _, err = d.DB.Exec(query); err != nil {
+		return fmt.Errorf("executing create schema: %w", err)
+	}
+	return nil
+}
+
+func (d *Deltalake) dropTable(table string) error {
+	var (
+		query string
+		err   error
+	)
+
+	query = fmt.Sprintf(`DROP TABLE %[1]s.%[2]s;`, d.Namespace, table)
+
+	if _, err = d.DB.Exec(query); err != nil {
+		return fmt.Errorf("executing drop table: %w", err)
+	}
+	return nil
+}
+
+func (d *Deltalake) dropStagingTables(stagingTables []string) {
+	for _, stagingTable := range stagingTables {
+		if err := d.dropTable(stagingTable); err != nil {
+			d.Logger.Warnw("failed dropping staging table",
+				logfield.SourceID, d.Warehouse.Source.ID,
+				logfield.SourceType, d.Warehouse.Source.SourceDefinition.Name,
+				logfield.DestinationID, d.Warehouse.Destination.ID,
+				logfield.DestinationType, d.Warehouse.Destination.DestinationDefinition.Name,
+				logfield.WorkspaceID, d.Warehouse.WorkspaceID,
+				logfield.Namespace, d.Namespace,
+				logfield.StagingTableName, stagingTable,
+				logfield.Error, err.Error(),
+			)
+		}
+	}
+}
```

Comment:
Won't it be better to create a single `Drop` query with all tables that are needed to be dropped, instead of doing it this way?

Reply:
Databricks don't support running drop table query with multiple [tables](https://docs.databricks.com/sql/language-manual/sql-ref-syntax-ddl-drop-table.html).